### DISCUSSION TITLE
Could @ignore be improved for relations?

### DISCUSSION THREAD

**Discussion by metafab** [prisma/prisma#21244] [Q&A]
### Question

For some reason, Prisma requires boths sides of a relation, even though one could be enough.

In the example provided below, the `user` property is useful to navigate from a PowerUser to the underlying BaseUser. However, the `powerUsers` property is useless and even misleading as there is only one PowerUser record for a given BaseUser record.

If the requirement to declare both sides cannot be removed, could `@ignore` be used to not have the `powerUsers` property generated?
Currently, adding `@ignore` to `BaseUser.powerUsers` results in having `PowerUser.user` to also not be generated, which is wrong. Why are both sides ignored when we want to exclude only one?

### Expected behavior (optional)

Either allow to declare only one side of a relation, or fix `@ignore` to ignore only one side, or allow both of this.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
model BaseUser {
  name String

  powerUsers PowerUser[]
  anotherKindOfUsers AnotherKindOfUser[]
}

model PowerUser {
  userId String
  someOtherProp String

  user BaseUser @relation(references: [id], fields: [userId])
}

model AnotherKindOfUser {
  userId String

  user BaseUser @relation(references: [id], fields: [userId])
}
```

**Comment by ludralph** [comment]
Hi @metafab 👋 

Thank you for raising this question.

Prisma indeed requires both sides of a relation to be declared. This is due to the way Prisma handles relations and ensures data integrity. There's currently no way to ignore only one side of a relation without also ignoring the other side. However, if you do want to see this added, can you please open a[ feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) and document your use case so that our engineers can take a look?

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nikolasburk** [reply]
@metafab you might also want to check out this recent discussion which was very similar and has additional context from our Product Manager who's responsible for Prisma ORM: https://github.com/prisma/prisma/discussions/21380

**Reply by metafab** [reply]
Thank you @nikolasburk. This is indeed highly related. Maybe having `@Ignore` work with relations would be a part of the solution.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
